esphome/components/psram/__init__.py

```diff
@@ -4,15 +4,35 @@
 from esphome.core import CORE
 from esphome.const import (
     CONF_ID,
+    CONF_MODE,
+    CONF_SPEED,
 )
 
 CODEOWNERS = ["@esphome/core"]
 
 psram_ns = cg.esphome_ns.namespace("psram")
 PsramComponent = psram_ns.class_("PsramComponent", cg.Component)
 
+SPIRAM_MODES = {
+    "quad": "CONFIG_SPIRAM_MODE_QUAD",
+    "oct": "CONFIG_SPIRAM_MODE_OCT",
+}
+
+SPIRAM_SPEEDS = {
+    "40m": "CONFIG_SPIRAM_SPEED_40M",
+    "80m": "CONFIG_SPIRAM_SPEED_80M",
+    "120m": "CONFIG_SPIRAM_SPEED_120M",
+}
+
 CONFIG_SCHEMA = cv.All(
-    cv.Schema({cv.GenerateID(): cv.declare_id(PsramComponent)}), cv.only_on_esp32
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(PsramComponent),
+            cv.Optional(CONF_MODE): cv.enum(SPIRAM_MODES, lower=True),
+            cv.Optional(CONF_SPEED): cv.enum(SPIRAM_SPEEDS, lower=True),
```

Comment:
This combine with the above change might be a little nicer
```suggestion
            cv.Optional(CONF_SPEED): cv.all(cv.frequency, cv.one_of(*SPIRAM_SPEEDS)),
```